spec/helpers/vm_helper_spec.rb

```diff
@@ -0,0 +1,23 @@
+require "spec_helper"
+
+# @record.operating_system && @record.operating_system.processes.maximum(:updated_on)
+
+describe VmHelper do
+  describe "#last_date_processes" do
+    it "supports vm without an OS" do
+      server  = FactoryGirl.build(:miq_server, :id => 99999)
+      operating_system = FactoryGirl.build(:operating_system)
+      @record = FactoryGirl.create(:vm_vmware, :miq_server => server, :operating_system => operating_system)
+      now = Time.now
+      operating_system.processes.create(:updated_on => 1.day.ago)
+      operating_system.processes.create(:updated_on => now)
+      operating_system.processes.create(:updated_on => 2.day.ago)
+      expect(helper.last_date_processes).to eq(now)
```

Comment:
or maybe you have to trim the fractional seconds by using `be within` or something